src/WordPress/WordPressClientAPI.php

```diff
@@ -115,4 +115,102 @@ public function createPageRuleDisablePerformanceCacheBypassJsonBody($urlPattern)
+
+            $url = add_query_arg($request->getParameters(), $this->getEndpoint().$request->getUrl());
+
+            $requestParams = array(
+                'timeout'=>30,
+                'method'=>$request->getMethod(),
+                'headers'=>$request->getHeaders()
+            );
+
+            $isPaginatable = false;
+            if ($requestParams['method'] === 'GET') {
+                $isPaginatable = true;
+            }
+
+            $mergedResponse = null;
+
+            $currentPage = 1;
+            $totalPages = 1;
+
+            while ($totalPages >= $currentPage) {
+
+                // Enable pagination
+                if ($isPaginatable) {
+                    $params['page'] = $currentPage;
+                }
+
+                if ($requestParams['method'] !== 'GET') {
+                    $requestParams['body'] = json_encode($request->getBody());
+                    $requestParams['headers']['Content-Type'] = 'application/json';
+                }
+
+                $requestResponse = wp_remote_request($url, $requestParams);
+
+                //check for connection error
+                if( is_wp_error($requestResponse) ) {
+                    $errorMessage = $requestResponse->get_error_message();
+
+                    $this->logAPICall($this->getAPIClientName(), array_merge(array( 'type' => 'request','path' => $url ), $requestParams), true);
+                    $this->logAPICall($this->getAPIClientName(), array('type' => 'response', 'reason' => $requestResponse->get_error_message(), 'code' =>  $requestResponse->get_error_code(), 'body' => $errorMessage), true);
+
+                    return $this->createAPIError($errorMessage);
+                }
+
+                //check for response error != 2XX
+                if( wp_remote_retrieve_response_code($requestResponse) > 299 ) {
+                    $errorMessage = wp_remote_retrieve_response_message($requestResponse);
+
+                    $this->logAPICall($this->getAPIClientName(), array_merge(array( 'type' => 'request','path' => $url ), $requestParams), true);
+                    $this->logAPICall($this->getAPIClientName(), array('type' => 'response', 'reason' => $errorMessage, 'code' =>  wp_remote_retrieve_response_code($requestResponse)), true);
+
+                    return $this->createAPIError($errorMessage);
+                }
+
+                $response = json_decode( wp_remote_retrieve_body($requestResponse), true );
+
+                if (json_last_error() !== JSON_ERROR_NONE) {
+                    $errorMessage = 'Error decoding client API JSON';
+                    $this->logAPICall($errorMessage, array( 'error' => json_last_error() ), true);
+
+                    return $this->createAPIError($errorMessage);
+                }
+
+                if (!$this->responseOk($response)) {
+                    $this->logAPICall($this->getAPIClientName(), array('type' => 'response', 'body' => $response), true);
+                }
```

Comment:
This is an issue from our code but it should return an error when `$this->responseOk($response)` is false. It's not a bug since `$mergedResponse = $response;` will be called and it'll return the same result but returning right away would make the code more readable.